src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -68,7 +81,12 @@ private static ElementAttributes fromEntity(DirectoryElementEntity entity) {
     }
 
     public Mono<Void> deleteElement(String elementUuid) {
-        return Mono.fromRunnable(() -> directoryElementRepository.deleteById(UUID.fromString(elementUuid)));
+        return Mono.fromRunnable(() -> deleteElementTree(elementUuid));
+    }
+
+    private void deleteElementTree(String elementUuid) {
```

Comment:
Maybe provide a parameter to enable/disable recursive deletion ? User might want to be notified when he is trying to remove a directory containing sub-directories.